### ISSUE TITLE
Serialization Issues after updating from 2.1.x to 2.2.x

### ISSUE THREAD

**Issue by crtl** [PAYONE-GmbH/magento-2#181]
Hello,
ive upgraded from 2.1.11 to 2.2.4 but the serialization was not correctly migrated.
Ive got an exception telling me that values cant be unserialized because payone configs are stored serialized and not in json format.

Ive already implemented my custom fix which is a bit insecure workaround:

```php
<?php
namespace Vendor\Module\Framework\Serialize\Serializer;

/**
 * Class JsonAdapterSerializer
 * Implements custom unserialize method which attempts to unserialize a serialized value before throwing exception
 * @package Vendor\Module\Framework\Serialize\Serializer
 */
class JsonAdapterSerializer extends \Magento\Framework\Serialize\Serializer\Json
{

    /**
     * {@inheritDoc}
     * @since 100.2.0
     */
    public function unserialize($string)
    {
        $result = json_decode($string, true);
        if (json_last_error() !== JSON_ERROR_NONE) {
            $result = unserialize($string, ["allowed_classes" => false]);
            if ($result === false) {
                throw new \InvalidArgumentException('Unable to unserialize value.');
            }
        }
        return $result;
    }
}
```

In `app/etc/di.xml` ive added the following and run `setup:di:compile` again.
```xml

<preference 
    for="Magento\Framework\Serialize\SerializerInterface" 
    type="Vendor\Module\Framework\Serialize\Serializer\JsonAdapterSerializer" />
```

**Comment by fjbender** [comment]
Basically, this should've been fixed with PR #144. I'm wondering why this doesn't work in your case.

**Comment by hreinberger** [comment]
closing this due to non-activity. If this is still an issue, please re-open or create a new one.
